### PR TITLE
Add com.polyphone_soundfonts.polyphone

### PR DESCRIPTION
**Pull request by hfiguiere**

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

File: com.polyphonesoundfonts.polyphone.json [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          "dest-filename": "polyphone-2-2.zip"
+        },
+        {
+          "type": "patch",
+          "path": "polyphone-make.patch",
+          "strip-components": 2
+        },
+        {
+          "type": "file",
+          "path": "com.polyphonesoundfonts.polyphone.appdata.xml"
```
Comment (LeandroStanger) [inline com.polyphonesoundfonts.polyphone.json]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

Comment (hfiguiere) [reply com.polyphonesoundfonts.polyphone.json]: done.

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

File: com.polyphonesoundfonts.polyphone.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  "sdk": "org.kde.Sdk",
+  "command": "polyphone",
+  "rename-desktop-file": "polyphone.desktop",
+  "rename-icon": "polyphone",
+  "finish-args": [
+    /* X11 + XShm access */
+    "--share=ipc",
+    "--socket=x11",
+    /* Wayland */
+    /* "--socket=wayland", */
```
Comment (barthalion) [inline com.polyphonesoundfonts.polyphone.json]: Just drop it if you don't need it.

**Comment by barthalion** [comment]
Hey @davy7125, would you interested in co-maintaining your app on Flathub?

File: com.polyphonesoundfonts.polyphone.json (hunk 2 above)
Comment (hfiguiere) [reply com.polyphonesoundfonts.polyphone.json]: I forget why it's commented out but it works with it in. So I'll add it back.

**Comment by hfiguiere** [comment]
Also I'm getting on the console a lot of messages:

```
QSocketNotifier: Can only be used with threads started with QThread
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../../src/pcm/pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib ../../../src/pcm/pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib ../../../src/pcm/pcm.c:2564:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_route_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_route_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_route_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_route_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_route_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_route_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_route_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_route_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_hooks_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_hooks_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_hooks_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_hooks_open is not defined inside [builtin]
ALSA lib ../../oss/pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib ../../oss/pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_plug_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_plug_open is not defined inside [builtin]
ALSA lib ../../usb_stream/pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib ../../usb_stream/pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_dmix_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_dmix_open is not defined inside [builtin]
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib ../../../src/pcm/pcm.c:8432:(snd_pcm_recover) underrun occurred
ALSA lib ../../../src/pcm/pcm.c:8432:(snd_pcm_recover) underrun occurred
ALSA lib ../../../src/pcm/pcm.c:8432:(snd_pcm_recover) underrun occurred
```
These last three line continue on.

**Comment by hfiguiere** [comment]
This doesn't change the continuous output of
```
ALSA lib ../../../src/pcm/pcm.c:8432:(snd_pcm_recover) underrun occurred
```

But it adds network access that seems to be need to access the sample library.

**Comment by hfiguiere** [comment]
This work but I'm not sure adding Jack support was the solution.

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
It works without jack.

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by BrunoVernay** [comment]
I started building a flatpak too! Glad I saw your post. Next time I will search GitHub first.

**Comment by BrunoVernay** [comment]
By the way, Jack 1.9.14 "_Use python3 for the waf build scripts_"
https://github.com/jackaudio/jack2/releases

**Comment by hfiguiere** [comment]
I still get the underruns in some circumstance and I have no idea why.

polyphone does NOT need jack to run and produce sound. It default to alsa and that works.

**Comment by hfiguiere** [comment]
You can have less jack errors `flatpak run --device=shm com.polyphonesoundfonts.polyphone` (if you have flatpak 1.6.1 or later)

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by zicklag** [comment]
Trying to install it with the command from the last successful build yieds:

    error: The application com.polyphonesoundfonts.polyphone/x86_64/test requires the runtime org.kde.Platform/x86_64/5.14 which was not found

I'm running Pop!_OS which uses gnome. Does it only work on KDE or something?

**Comment by hfiguiere** [comment]
it just needs the kde runtime from flathub. I use GNOME too. Sadly for test builds it is not automatic.

Something like

`flatpak install flathub org.kde.Platform//5.14`

**Comment by BrunoVernay** [comment]
Seems to be working. There are still a few error messages, but no crashes.

```
flatpak run --user -v  --device=shm  com.polyphonesoundfonts.polyphone            
F: Opening user flatpak installation at path /home/bruno2/.local/share/flatpak
F: Opening user flatpak installation at path /home/bruno2/.local/share/flatpak
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening user flatpak installation at path /home/bruno2/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Regenerating ld.so.cache /home/bruno2/.var/app/com.polyphonesoundfonts.polyphone/.ld.so/f9edeb333113246192eeda24e71d978dcfe36897163adb8c2b41820da13bef7f
F: Running: 'bwrap --args 10 ldconfig -X -C /run/ld-so-cache-dir/f9edeb333113246192eeda24e71d978dcfe36897163adb8c2b41820da13bef7f'
ldconfig: /app/lib/libstk-4.6.1.so is not a symbolic link

F: Allocated instance id 2223532005
F: Add defaults in dir /com/polyphonesoundfonts/polyphone/
F: Add locks in dir /com/polyphonesoundfonts/polyphone/
F: Allowing homedir access
F: Allowing wayland access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/bruno2/.config/pulse/client.conf': Error opening file /home/bruno2/.config/pulse/client.conf: No such file or directory
F: Running 'bwrap --args 31 xdg-dbus-proxy --args=33'
F: Running 'bwrap --args 31 polyphone'
QSocketNotifier: Can only be used with threads started with QThread

(process:2): Gtk-WARNING **: 18:42:16.832: Locale not supported by C library.
	Using the fallback 'C' locale.
ALSA lib ../../src/dlmisc.c:170:(snd_dlsym_verify) unable to verify version for symbol _snd_pcm_empty_open
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_empty_open is not defined inside [builtin]
... Lots of ALSA lib errors ....
ALSA lib ../../src/dlmisc.c:295:(snd1_dlobj_cache_get) symbol _snd_pcm_dmix_open is not defined inside [builtin]
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock

RtMidiIn: no compiled support for specified API argument!
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()


```

If I start Jack there are even more error messages.

**Comment by hfiguiere** [comment]
OK this is odd and the more it goes, the more I think my system has some issues.

I was trying to figure out why the underruns and such, and suddenly I no longer get them.

**Comment by hfiguiere** [comment]
And I plug my headset, and I get the underruns.

(I don't get them with a non flatpak build)

**Comment by hfiguiere** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
This package work on my system. There was something weird.

The last changes also make sure that jack is buillt last as we don't support Jack (but it's required to build)

**Comment by hfiguiere** [comment]
bot, build com.polyphone_soundfonts.polyphone

**Comment by zicklag** [comment]
Hey, this just worked for me. :smiley: I wasn't able to test the last dev build because the artifact expired or something.

Is there anything keeping this from going into the store?

**Comment by hfiguiere** [comment]
> Hey, this just worked for me. smiley I wasn't able to test the last dev build because the artifact expired or something.

Sadly the test build do expire. But a new one can be retriggered.

> Is there anything keeping this from going into the store?

It's good on my side. @bilelmoussaoui (since you put the label) ?

**Comment by bilelmoussaoui** [comment]
bot, build com.polyphonesoundfonts.polyphone

**Comment by hfiguiere** [comment]
bot, build com.polyphone_soundfonts.polyphone

**Pull request closed without merging**